src/sinks/splunk_hec.rs

```diff
@@ -296,6 +294,28 @@ fn build_uri(host: &str, path: &str) -> Result<Uri, http::uri::InvalidUri> {
     format!("{}{}", host.trim_end_matches('/'), path).parse::<Uri>()
 }
 
+#[derive(Debug, Copy, Clone)]
+struct RequestByteSize(usize);
+
+impl From<&hyper::Request<Vec<u8>>> for RequestByteSize {
+    fn from(request: &hyper::Request<Vec<u8>>) -> Self {
+        RequestByteSize(request.body().len())
+    }
+}
+
+impl crate::sinks::util::sink::Response for (hyper::Response<bytes::Bytes>, RequestByteSize) {
+    fn is_successful(&self) -> bool {
+        self.0.status().is_success()
```

Comment:
Does this resolve the user reported errors that Vector fails to retry requests?

Reply:
No, this does not change retry request logic. Currently, we have shared `Response` trait implementation: https://github.com/timberio/vector/blob/1907db610b112ca275387b70f1326243e55c4c39/src/sinks/util/http.rs#L371-L375
with this PR, we will need to implement `Response` for every sink because `emit_events` function will emit different events.